internal/services/jsonrpc/usersvc/service.go

```diff
@@ -42,6 +43,14 @@ type Service struct {
 	chainClient BlockchainTransactor
 	pricer      Pricer
 	migrator    Migrator
+
+	// authenticate RPCs?
+	authenticateRPCs bool
+	// challenges issued to the clients
+	// TODO: have a goroutine that cleans up expired challenges for every few hrs?
```

Comment:
I don't think we should punt on this as long as we are using this approach of allowing unlimited challenge requests.

Reply:
Yeah agreed.

Reply:
+1. Also, this map can be called concurrently, so you should use `internal/utils/sync_map`

Reply:
Ah, nvm I see you have a mutex. Syncmap hides this / ensures that we don't forget it, but I'll leave it up to you to decide.